### PR TITLE
[NT-946] Personalization/onboarding experiment logic

### PR DESCRIPTION
**Pull request by ifbarrera**
# 📲 What

Adds the logic in the `AppDelegateViewModel` for displaying the personalization/onboarding screen if the user is eligible and in the experimental variant for the experiment.

# 🤔 Why

Since this onboarding flow will be run as an experiment, we want to check whether the user is in the experimental variant and only then show the onboarding screen. Additionally, the onboarding screen should only be shown once.

# 🛠 How

- added an output for showing the onboarding/personalization screen
- logic checks user defaults or previously set flag, and then checks Optimizely to get the user's experiment variant
- all deeplinks triggered via notification/handoff/shortcut/universal links are intercepted. If the user has not seen the onboarding page but is eligible to see it, the deeplink will not resolve in order to allow the onboarding page to show

# ✅ Acceptance criteria

In `Service.swift`, set the default serverConfig value to `.staging`. This is required because when the app launches clean for the first time, it needs to be set to staging so that we can access the staging Optimizely configuration.

image

- [x] on a fresh build (no previous build installed on device/sim), run the app. You should see the category selection screen presented after a short moment. If you run the app again, the screen should not display.

### CODE REVIEW

File: Library/OptimizelyExperiment.swift [hunk 1]
```diff
@@ -3,6 +3,7 @@ import Foundation
 public enum OptimizelyExperiment {
   public enum Key: String {
     case pledgeCTACopy = "pledge_cta_copy"
+    case categoryPersonalization = "onboarding_category_personalization_flow"
```
Comment (justinswart) [inline Library/OptimizelyExperiment.swift]: With our other experiments we've camelCased the snake_cased value as the case name.

File: Kickstarter-iOS/ViewModels/AppDelegateViewModel.swift [hunk 2]
```diff
@@ -1087,3 +1094,31 @@ private func qualtricsConfigData() -> QualtricsConfigData {
...
+  let hasSeenCategoryPersonalization = AppEnvironment.current.userDefaults.hasSeenCategoryPersonalizationFlow
+
+  if isLoggedIn || hasSeenCategoryPersonalization {
+    // Currently logged-in users should not see the onboarding flow
+    AppEnvironment.current.userDefaults.hasSeenCategoryPersonalizationFlow = true
+
+    return false
+  }
+
+  let variant = AppEnvironment.current.optimizelyClient?.variant(
```
Comment (justinswart) [inline Kickstarter-iOS/ViewModels/AppDelegateViewModel.swift]: So this will still load async and we may briefly see the discover feed?

Comment (ifbarrera) [reply Kickstarter-iOS/ViewModels/AppDelegateViewModel.swift]: That's right!

File: Library/OptimizelyExperiment.swift (hunk 1 above)
Comment (ifbarrera) [reply Library/OptimizelyExperiment.swift]: It's a little verbose, but ok.

**Comment by nativeksr** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Pull request merged**